api/v1alpha4/virtualmachine_types.go

```diff
@@ -895,6 +896,29 @@ type VirtualMachineSpec struct {
 	// virtual machine. These settings take effect during the next power-on of the
 	// virtual machine.
 	BootOptions *VirtualMachineBootOptions `json:"bootOptions,omitempty"`
+
+	// +optional
+
+	// CurrentSnapshot represents the desired snapshot that the
+	// virtual machine should point to. This field can have three
+	// possible values:
+	//
+	// The value of this field is nil when the working snapshot is at
+	// the root of the snapshot tree.
+	//
+	// When a new snapshot is requested by creating a new
+	// VirtualMachineSnapshot, the value of this field is set to the
+	// new snapshot resource's name.
+	//
+	// If the value of this field is set to an existing snapshot that
+	// is different from the status.currentSnapshot, the virtual machine is
+	// reverted to the requested snapshot. A revert operation results
+	// in the virtual machine's configuration and data being rolled
+	// back to the state of the snapshot. When other fields of the
+	// VirtualMachineSpec (e.g., PowerState) are updated along with
+	// the revert operation, then those are applied once the revert is
+	// completed.
+	CurrentSnapshot *corev1.TypedLocalObjectReference `json:"currentSnapshot,omitempty"`
```

Comment:
When should we use `corev1.TypedLocalObjectReference` vs `vmopv1common.LocalObjectRef`? (Asking as I'm using the latter in #989 ) They both have the same Kind+Name fields, only diff is `APIGroup *string` vs `APIVersion string`.

Reply:
Yeah, so apparently there was a somewhat recent guidance recommending *not* to use the corev1 types.
https://github.com/kubernetes/api/commit/1aa9f66df7d5e3c79de1b0c157882dac343838ea

Until now, this warning was only present on corev1.ObjectReference but upstream has not added it on _all_ types (localref, localobjectref etc).

Let's use our internal one.